tests/Unit/Domain/Test_ElementId.cpp

```diff
@@ -19,6 +19,64 @@
 #include "Utilities/StdHelpers.hpp"  // IWYU pragma: keep
 
 namespace {
+template <size_t VolumeDim>
+void test_placement_new_and_hashing_impl(
+    const size_t block1, const std::array<SegmentId, VolumeDim>& segments1,
+    const size_t block2, const std::array<SegmentId, VolumeDim>& segments2) {
+  using Hash = std::hash<ElementId<VolumeDim>>;
+
+  const ElementId<VolumeDim> id1(block1, segments1);
+  const ElementId<VolumeDim> id2(block2, segments2);
+
+  ElementId<VolumeDim> element_index1{}, element_index2{};
```

Comment:
Naming these `ElementId`s as `element_index1,2` is confusing. Could these be `test_id1,2`? or `element_id1,2`?